src/exes/moc.ml

```diff
@@ -46,8 +47,11 @@ let argspec = [
       Arg.String (fun fp -> Flags.post_ref := Some fp);
       Arg.Unit (set_mode StableCompatible);
       ],
-   "<pre> <post> test upgrade compatibility between stable-type signatures <pre> and <post>";
-  "--idl", Arg.Unit (set_mode Idl), " generate IDL spec";
+    "<pre> <post> test upgrade compatibility between stable-type signatures <pre> and <post>";
+  "--idl", Arg.Unit (fun () ->
+    idl := true;
+    set_mode Compile ()), (* HACK to support dfx 0.8.3 or earlier *)
```

Comment:
Why is this a hack? We need to set `moc` to compile mode for `--idl`. 

Is there a flag check to disallow `--idl` in interpreter or wasi mode?

Reply:
Well, it's a hack in make the first invocation of old dfx versions to work, by forcing compilation

Reply:
But I can remove the comment...